packages/core/upload/server/register.js

```diff
@@ -60,6 +62,25 @@ const createProvider = (config) => {
     );
   }
 
+  if (providerOptions?.sizeLimit) {
+    // TODO V5: remove sizeLimit from providerOptions
+    process.emitWarning(
+      `[deprecated] In future versions, "sizeLimit" argument will be ignored from upload.config.providerOptions. Move it to upload.config`
+    );
+  }
+
+  if (!providerInstance.checkFileSize) {
+    providerInstance.checkFileSize = (file) => {
+      const fileSize = kbytesToBytes(file.size);
+
+      if (providerOptions?.sizeLimit && fileSize > providerOptions.sizeLimit) {
```

Comment:
that if actually should be kept inside the local provider code as it's not a shared option at the moment